### PR TITLE
Code corrections and Update Server via Github

### PR DESCRIPTION
**Pull request by twitnic**
Corrected some Codestyle and added update_server.xml. So Joomla is checking updates via github.

Joomla -> search on https://raw.githubusercontent.com/jemproject/JEM-Project/master/update_server.xml -> this is xml in project with version informations -> when new version is released you must update the version-number in update_server.xml and joomla checks as new Update

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -167,3 +167,50 @@ pip-log.txt
 #############
 /build.config
 /build.number
+
+### JetBrains template
```
Comment (twitnic) [inline .gitignore]: I have PhpStorm, so I must added it.

File: site/classes/activecalendarweek.php [hunk 2]
```diff
@@ -20,14 +20,17 @@
 
 require_once('calendar.class.php');
 
-class ActiveCalendarWeek extends JEMCalendar {
+class ActiveCalendarWeek extends JemCalendar {
```
Comment (twitnic) [inline site/classes/activecalendarweek.php]: classes are case sensitive

File: site/classes/activecalendarweek.php [hunk 3]
```diff
@@ -20,14 +20,17 @@
...
 	/*
 	********************************************************************************
 	Names of the generated html classes. You may change them to avoid any conflicts with your existing CSS
 	********************************************************************************
 	*/
-	var $cssWeeksTable="week";
-	var $cssMonthWeek="monthweek";
+	public $cssWeeksTable = "week";
```
Comment (twitnic) [inline site/classes/activecalendarweek.php]: In php4 it was var. Since php5 it is public / private ...

File: site/classes/activecalendarweek.php [hunk 4]
```diff
@@ -49,7 +52,7 @@ public function __construct($year=false, $month=false, $day=false, $GMTDiff="non
 	It generates as many rows as set in $numberOfWeeks
 	********************************************************************************
 	*/
-	function showWeeks($numberOfWeeks=1) {
+	public function showWeeks($numberOfWeeks = 1) {
```
Comment (twitnic) [inline site/classes/activecalendarweek.php]: Class properties must be defined as public, private, or protected. If declared using var or without any declaration, the property will be defined as public.

File: site/classes/attachment.class.php [hunk 5]
```diff
@@ -13,7 +13,7 @@
 jimport('joomla.filesystem.folder');
 
 // ensure JemFactory is loaded (because this class is used by modules or plugins too)
-require_once(JPATH_SITE.'/components/com_jem/factory.php');
+require_once(JPATH_SITE . '/components/com_jem/factory.php');
```
Comment (twitnic) [inline site/classes/attachment.class.php]: please use the joomla style guide. So other user and tools (code sniffer ... ) could make better code.

File: site/classes/attachment.class.php [hunk 6]
```diff
@@ -138,18 +145,20 @@ static function update($attach)
 
 	/**
 	 * return attachments for objects
-	 * @param string object identification (should be event<eventid>, category<categoryid>, etc...)
+	 * @param string $object identification (should be event<eventid>, category<categoryid>, etc...)
+	 *
+	 * @since 1.1
```
Comment (twitnic) [inline site/classes/attachment.class.php]: In Joomla it is a must, to add the return and the since tag.

File: update_server.xml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        </description>
+        <element>com_jem</element>
+        <type>component</type>
+        <version>2.2.0</version>
+        <infourl title="JED - Joomla Event Manager">
+            http://www.joomlaeventmanager.net
+        </infourl>
+        <downloads>
+            <downloadurl type="full" format="zip">
+                https://github.com/jemproject/JEM-Project/archive/v2.2.0-stable.zip
```
Comment (twitnic) [inline update_server.xml]: On a new Update you must at first upload the new zip. Copy the URL of this and paste it in the xml. than push this xml with a new version string.

File: jem.xml [hunk 8]
```diff
@@ -87,8 +87,7 @@
 		</files>
 	</administration>
 
-	<!--updateservers-->
-		<!--  TODO: Fix URL when feature is ready to use -->
-		<!--server type="extension" priority="1" name="JEM Update Site">http://www.joomlaeventmanager.invalid/update/jem-update.xml</server-->
-	<!--/updateservers-->
+	<updateservers>
+		<server type="extension" priority="1" name="JEM Update Site">https://raw.githubusercontent.com/jemproject/JEM-Project/master/update_server.xml</server>
```
Comment (twitnic) [inline jem.xml]: The direct link to the xml in the master branch.

Comment (twitnic) [inline jem.xml]: bildschirmfoto 2016-09-20 um 18 36 30

File: update_server.xml (hunk 7 above)
Comment (twitnic) [inline update_server.xml]: #77 added update-implementation for Joomla 3.
`<targetplatform name="joomla" version="3.*"/>`

**Comment by Hoffi1** [comment]
Sorry, but I do not approve to merge this PR. It doesn't give any benefit to beautify all files without any functional or real optical effect. Instread it makes it impossible to see real modifications between previous and later versions. And it would require a lot of work to correct the wrong indentations using spaces instead of tabs we expect.

btw: The reference to the zip file on GitHub for update is wrong because this file will not update the complete package which results in inconsistent sites.

**Comment by twitnic** [comment]
And you would not start to get the coding guidelines into the component? Okaaaay ...

**Comment by twitnic** [comment]
Impossible for good coding is a project with a mass of wrong PhpDocs, Declarations , etc ..

I just want help ...

**Comment by diesl** [comment]
Hi Twitnic,

as I stated in our previous conversation via PM in the JEM forum, we do appreciate help and especially developers who want to improve JEM. It's just about the "how".

The codestyle is a good idea in general, but as I told you, a discussion in the forum before this PR would have been a better idea. We need to reach a consensus on this topic with every developer, because it possibly affects every file.

Some aspects:
- We prefer not to change codestyle for the sole purpose of codestyling. Instead fix codestyle when changing the code anyway.
- We possibly want to make some minor changes to the standard Joomla codestyle
- You mixed up two topics in this PR: update server and codestyle

Sorry for you having put time into this topic. I hope we come to a conclusion that helps everybody. I am sure, with a little planning we can avoid these little controversies in the future :+1:

**Comment by twitnic** [comment]
In the next step I have wrote unit tests. So you could test, is the PR without errors, than merge it. Otherwise not. And so you find fast bugs in your code.

So ... I close the PR ...

**Pull request closed without merging**